File: src/paths.py
"""Single source of truth for where runtime data lives.

Everything the app writes — briefing.json, chat sessions, health logs,
dialectics, scheduled messages, agent state, webhook state — sits under one
directory. That directory is normally <repo>/data, but `DB_DATA_DIR` overrides
it.

The override exists so tests never touch real data. Point DB_DATA_DIR at a temp
directory and the whole app — the briefing build, the MCP server, the chat
handler, the webhook — reads and writes there instead, so a test run cannot
delete or truncate a live chat session, health log, or dialectic. There is no
"clean up afterwards" step to get wrong: throw the temp directory away.

The PHP half of the app resolves the same variable in web/paths.php; the two
must agree, since web/chat.php and web/webhook.php spawn the Python handler and
pass DB_DATA_DIR through to it.
"""
import os

BASE_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))


def data_dir():
    """Absolute path to the runtime data directory. Honours DB_DATA_DIR."""
    override = os.environ.get('DB_DATA_DIR', '').strip()
    if override:
        return os.path.abspath(os.path.expanduser(override))
    return os.path.join(BASE_DIR, 'data')


def data_path(*parts):
    """Path to something inside the data directory."""
    return os.path.join(data_dir(), *parts)


def ensure_data_dir(*parts):
    """Like data_path, but creates the containing directory first."""
    path = data_path(*parts)
    parent = path if not parts else os.path.dirname(path)
    os.makedirs(parent, exist_ok=True)
    return path

File: web/paths.php
<?php
// PHP half of the data-directory resolver. Mirrors src/paths.py — keep the two
// in sync; both must honour DB_DATA_DIR or a test that overrides it in one
// language will still write to real data through the other.
//
// chat.php and webhook.php spawn the Python handler, so they also pass
// DB_DATA_DIR down to it (see db_run_agent in agent_client.php).

/**
 * Absolute path to the runtime data directory. DB_DATA_DIR overrides the
 * default <repo>/data, so tests can be pointed at a throwaway directory and
 * physically cannot touch live sessions, health logs, or dialectics.
 */
function db_data_dir()
{
    $override = getenv('DB_DATA_DIR');
    if ($override === false || trim($override) === '') {
        // $_SERVER carries it under some SAPIs where getenv() does not.
        $override = isset($_SERVER['DB_DATA_DIR']) ? $_SERVER['DB_DATA_DIR'] : '';
    }
    $override = trim((string)$override);
    if ($override !== '') {
        return rtrim($override, '/');
    }
    return dirname(__FILE__) . '/../data';
}

/** Path to something inside the data directory. */
function db_data_path($rel)
{
    return db_data_dir() . '/' . ltrim($rel, '/');
}
